File: easy/4024.py
"""
4024. Nearest Available Drone

https://leetcode.com/problems/nearest-available-drone/

Weekly Contest 515
"""


class Solution:
    def nearestDrone(self, drones: list[list[int]], target: list[int]) -> int:
        target_x, target_y = target

        out: int = -1
        min_distance: int = 101

        for idx, (x, y, r) in enumerate(drones):
            distance: int = abs(x - target_x) + abs(y - target_y)

            if distance <= r and distance < min_distance:
                min_distance = distance
                out = idx

        return out

File: easy/4024.rs
// 4024. Nearest Available Drone
// https://leetcode.com/problems/nearest-available-drone/
// Weekly Contest 515

impl Solution {
    pub fn nearest_drone(drones: Vec<Vec<i32>>, target: Vec<i32>) -> i32 {
        let mut out: i32 = -1;
        let mut min_distance: i32 = i32::MAX;

        for (idx, drone) in drones.iter().enumerate() {
            let distance: i32 = (drone[0] - target[0]).abs() + (drone[1] - target[1]).abs();

            if distance <= drone[2] && distance < min_distance {
                min_distance = distance;
                out = idx as i32;
            }
        }

        out
    }
}
